### PR TITLE
[DeploymentScripts] Inclusion of ContainerGroupName

### PR DESCRIPTION
**Pull request by jorgecotillo**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json [hunk 1]
```diff
@@ -651,6 +656,16 @@
         }
       }
     },
+    "ContainerConfiguration": {
+      "type": "object",
+      "description": "Settings to customize ACI container instance.",
+      "properties": {
+        "containerGroupName": {
+          "type": "string",
+          "description":  "Container group name, if not specified then the name will get auto-generated."
```
Comment (filizt) [inline specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json]: nit - Maybe add some information about when to provide a containerGroupName?

**Comment by filizt** [comment]
LGTM

File: specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json (hunk 1 above)
Comment (majastrz) [reply specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json]: +1 to that.

Comment (majastrz) [reply specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json]: I would also make sure that we have an example with and without `containerGroupName`.

---
In reply to: [395333253](https://github.com/Azure/azure-rest-api-specs/pull/8771#discussion_r395333253) [](ancestors = 395333253)

**Review by majastrz — APPROVED** [review]
Signed off from ARM side. Please take a look at the comments, though.

File: specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json (hunk 1 above)
Comment (jorgecotillo) [reply specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json]: Sounds good. I'll add an example with an without containerGroupName, good suggestion.
Re: name rules, these are basically the same from ACI I'll make sure to add the rules as part of the description.
Thanks!

Comment (jorgecotillo) [reply specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json]: Additional info added. Thanks!

File: specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json [hunk 2]
```diff
@@ -651,6 +656,16 @@
         }
       }
     },
+    "ContainerConfiguration": {
+      "type": "object",
+      "description": "Settings to customize ACI container instance.",
+      "properties": {
+        "containerGroupName": {
+          "type": "string",
+          "description":  "Container group name, if not specified then the name will get auto-generated. Not specifying a 'containerGroupName' indicates the system to generate a unique name which might end up flagging an Azure Policy as non-compliant. Use 'containerGroupName' when you have an Azure Policy that expects a specific naming convention or when you want to fully control the name. 'containerGroupName' property must be between 1 and 63 characters long, must contain only lowercase letters, numbers, and dashes and it cannot start or end with a dash and consecutive dashes are not allowed. To specify a 'containerGroupName', add the following object to properties: { \"containerSettings\": { \"containerGroupName\": \"contoso-container\" } }. If you do not want to specify a 'containerGroupName' then do not add 'containerSettings' property."
```
Comment (filizt) [inline specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json]: You can use "maxLength": 63 here. I'd think you can specify minLength too.

Comment (jorgecotillo) [reply specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json]: Thanks Filiz, min and max added

**Pull request merged**